src/network/client.rs

```diff
@@ -145,29 +145,27 @@ impl Client {
 
 	async fn put_kad_record_batch(&self, records: Vec<Record>, quorum: Quorum) -> NumSuccPut {
 		let mut num_success: usize = 0;
+		let (tx, mut rx) = mpsc::channel(100);
 		for records in records.chunks(self.put_batch_size).map(Into::into) {
-			let (sender, receiver) = oneshot::channel();
 			if self
 				.sender
 				.send(Command::PutKadRecordBatch {
 					records,
 					quorum,
-					sender,
+					sender: tx.clone(),
 				})
 				.await
 				.context("Command receiver should not be dropped.")
```

Comment:
Out of a scope of this change, but in this case error is discarded, so context is not needed. Also, in case of send error, num_success will always be 0, since send is blocking. This is clearer if code is split:

```